### PR TITLE
[CHI-68] Add sliding border to tab component

### PR DESCRIPTION
**Pull request by dani-cl-madrid**
@mattnickles @jllr 

https://elasticbox.atlassian.net/browse/CHI-68

### CODE REVIEW

File: src/website/views/components/tabs.pug [hunk 1]
```diff
@@ -315,3 +316,94 @@ h4 Large
...
+    ul#example-tabs-2.a-tabs
+      li.-active
+        a(href='#a2') Tab a
+      li
+        a(href='#b2') Tab b
+      li
+        a(href='#c2') Tab c
+      li
+        a(href='/')(target='_self') Link
+      li Adios
```
Comment (mattnickles) [inline src/website/views/components/tabs.pug]: Can we remove the Adios list item

File: src/website/views/components/tabs.pug [hunk 2]
```diff
@@ -315,3 +316,94 @@ h4 Large
...
+    </div>
+    <div class="a-tabs-panel" id="c">
+      <p>Content for tab c</p>
+    </div>
+
+    <script src="https://assets.ctl.io/chi/0.8.2/tabs.js"></script>
+    <script>chiTabs.init(document.getElementById('example-tabs'));</script>
+
+h4 Keep default link behaviour
+p By default, Chi javascript enabled tabs will ignore their default anchor behaviour.
```
Comment (mattnickles) [inline src/website/views/components/tabs.pug]: paragraph needs `.-text` class

File: src/website/views/components/tabs.pug [hunk 3]
```diff
@@ -315,3 +316,94 @@ h4 Large
...
+      <p>Content for tab a</p>
+    </div>
+    <div class="a-tabs-panel" id="b">
+      <p>Content for tab b</p>
+    </div>
+    <div class="a-tabs-panel" id="c">
+      <p>Content for tab c</p>
+    </div>
+
+    <script src="https://assets.ctl.io/chi/0.8.2/tabs.js"></script>
```
Comment (mattnickles) [inline src/website/views/components/tabs.pug]: Can we make the tabs.js requirement more prominent by removing it from the two examples and placing it into it's own box right after the "Chi library provides a vanilla JavaScript..." paragraph.
```
.-mb2
  :code(lang='html')
    <script src="https://assets.ctl.io/chi/0.8.2/tabs.js"></script>
```

File: src/chi/components/tabs/tabs.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    if (tabPane) {
+      helpers.removeClass(tabPane, chiActiveClass);
+    }
+  };
+
+  let moveSlidingBorder = function (tab, tabsComponent) {
+    let offset = 0;
+    let size = 0;
+    let found = false;
+    const vertical = helpers.hasClass(tabsComponent, '-vertical');
```
Comment (mattnickles) [inline src/chi/components/tabs/tabs.js]: Can we add a vertical example to the documentation?

File: src/website/views/components/tabs.pug [hunk 5]
```diff
@@ -315,3 +316,94 @@ h4 Large
...
+      p Content for tab a
+    #b2.a-tabs-panel
+      p Content for tab b
+    #c2.a-tabs-panel
+      p Content for tab c
+  :code(lang='html')
+    <ul class="a-tabs" id="example-tabs">
+      <li class="-active"><a href="#a">Tab a</a></li>
+      <li><a href="#b">Tab b</a></li>
+      <li><a href="#c">Tab c</a></li>
```
Comment (mattnickles) [inline src/website/views/components/tabs.pug]: We're missing the default anchor link in the code snippet

File: src/website/views/components/tabs.pug [hunk 6]
```diff
@@ -315,3 +316,94 @@ h4 Large
...
+    </ul>
+
+    <div class="a-tabs-panel -active" id="a">
+      <p>Content for tab a</p>
```
Comment (mattnickles) [inline src/website/views/components/tabs.pug]: paragraphs in the examples need `.-text` class

File: src/chi/components/tabs/tabs.scss [hunk 7]
```diff
@@ -55,15 +63,22 @@
...
-        }
-
         & > a {
           line-height: $line-height-smaller;
           padding: 1.25rem 0;
         }
+
+        &.a-tabs__sliding-border {
+          margin-left: 0;
+          height: 0.25rem;
```
Comment (mattnickles) [inline src/chi/components/tabs/tabs.scss]: Can we change the sliding borders height value to 0.125rem (2px) and apply that not just here but on the non-js active tab as well?

File: src/website/views/components/tabs.pug [hunk 8]
```diff
@@ -315,3 +316,190 @@ h4 Large
         <a href="#">Tab Link</a>
       </li>
     </ul>
+
+
+h3 JavaScript
+p.-text
+| Chi library provides a vanilla JavaScript file that enables default tabs functionality.
```
Comment (jllr) [inline src/website/views/components/tabs.pug]: This must be indented

File: src/website/views/components/tabs.pug [hunk 9]
```diff
@@ -315,3 +316,190 @@ h4 Large
...
+
+.-mb2
+  :code(lang='html')
+    <script src="https://assets.ctl.io/chi/0.8.2/tabs.js"></script>
+
+p.-text
+| You have to link the tabs.js file and initialize every single tab panel you want to be activated.
```
Comment (jllr) [inline src/website/views/components/tabs.pug]: This must be indented

**Pull request merged**